src/averageAges.js

```diff
@@ -54,8 +80,25 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let totalAgeDiff = 0;
+  let motherCount = 0;
+
+  people.forEach((person) => {
+    people.forEach((mother) => {
+      if (onlyWithSon && person.sex === 'm' && person.mother === mother.name) {
+        totalAgeDiff += person.born - mother.born;
+        motherCount++;
+      }
+
+      if (!onlyWithSon && person.mother === mother.name) {
+        totalAgeDiff += person.born - mother.born;
+        motherCount++;
+      }
+    });
+  });
+
+  return totalAgeDiff / motherCount;
```

Comment:
Replace the forEach loop with a reduce function to calculate the average age difference. Example: `return ageDiffs.reduce((sum, ageDiff) => sum + ageDiff, 0) / ageDiffs.length;`